src/util/Helpers.cpp

```diff
@@ -316,6 +324,50 @@ QLocale getSystemLocale()
     return QLocale::system();
 }
 
+void writeProviderEmotesCache(const QString &id, const QString &provider,
+                              const QByteArray &bytes)
+{
+    std::ignore = QtConcurrent::run([bytes, id, provider]() {
+        QFile responseCache(getApp()->getPaths().cacheDirectory() + "/" + id +
+                            "." + provider);
+
+        if (responseCache.open(QIODevice::WriteOnly))
+        {
+            qCDebug(chatterinoCache)
+                << "Saved json response " << id << "." << provider;
+            responseCache.write(bytes);
```

Comment:
I think we should compress the data. Using the free function [`qCompress`](https://doc.qt.io/qt-6/qbytearray.html#qCompress) here (and `qUncompress` on the other side) caused the cached 7TV response of my channel to be just 5% of the original size.